src/interfaces/wallet.cpp

```diff
@@ -54,7 +54,7 @@ class PendingWalletTxImpl : public PendingWalletTx
 };
 
 //! Construct wallet tx struct.
-WalletTx MakeWalletTx(CWallet& wallet, const CWalletTx& wtx)
+WalletTx MakeWalletTx(CWallet& wallet, const CWalletTx& wtx) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
```

Comment:
a) annotations need to go in the header file, not the cpp file. b) I dont think we can put locks required in the interfaces file, as it may become a process boundary. The callee will have to take the lock itself.

Reply:
> annotations need to go in the header file

This is ok, these are just internal functions (in an anonymous namespace) not shared or exposed in headers.